### PR TITLE
automatically put nonlinear systems in canonical form

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Fixes https://github.com/SciML/ModelingToolkit.jl/issues/1037

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Unrelated CI error @YingboMa ?

File: src/systems/nonlinear/nonlinearsystem.jl [hunk 1]
```diff
@@ -65,7 +65,9 @@ function NonlinearSystem(eqs, states, ps;
                          systems=NonlinearSystem[],
                          connection_type=nothing,
                          )
-    eqs = collect(eqs)
```
Comment (YingboMa) [inline src/systems/nonlinear/nonlinearsystem.jl:68]: This should be kept.

File: src/systems/nonlinear/nonlinearsystem.jl [hunk 2]
```diff
@@ -65,7 +65,9 @@ function NonlinearSystem(eqs, states, ps;
...
+    # Move things over, but do not touch array expressions
+    eqs = eqs isa Vector{Equation} ? [0 ~ x.rhs - x.lhs for x in eqs] : eqs
```
Comment (YingboMa) [inline src/systems/nonlinear/nonlinearsystem.jl:69]: The branch looks unnecessary.

Comment (ChrisRackauckas) [reply src/systems/nonlinear/nonlinearsystem.jl:69]: it's not. `D.(x) .~ ...` symbolic array turns into symbolic broadcast expression. I'm just going to ignore that case for now by skipping it 😅

Comment (YingboMa) [reply src/systems/nonlinear/nonlinearsystem.jl]: That's why there is `collect`.

**Comment by YingboMa** [comment]
Maybe squash the commits next time?

**Comment by ChrisRackauckas** [comment]
yeah

**Pull request merged**